openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateFluidCoolerSingleSpeed.cpp

```diff
@@ -0,0 +1,147 @@
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#include "../ForwardTranslator.hpp"
+#include "../../model/FluidCoolerSingleSpeed.hpp"
+#include "../../model/FluidCoolerSingleSpeed_Impl.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../utilities/idd/IddEnums.hpp"
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/FluidCooler_SingleSpeed_FieldEnums.hxx>
+
+using namespace openstudio::model;
+
+namespace openstudio {
+
+namespace energyplus {
+
+boost::optional<IdfObject> ForwardTranslator::translateFluidCoolerSingleSpeed(FluidCoolerSingleSpeed & modelObject)
+{
+  IdfObject idfObject(openstudio::IddObjectType::FluidCooler_SingleSpeed);
+
+  m_idfObjects.push_back(idfObject);
+
+  OptionalString s;
+  OptionalDouble d;
+  OptionalModelObject mo;
+
+  ///////////////////////////////////////////////////////////////////////////
+  s = modelObject.name();
+  if (s)
+  {
+    idfObject.setName(*s);
+  }
+  ///////////////////////////////////////////////////////////////////////////
+
+  ///////////////////////////////////////////////////////////////////////////
+  mo = modelObject.waterInletNode();
+  if (mo)
+  {
+    s = mo->name();
+    if (s)
+    {
+      idfObject.setString(openstudio::FluidCooler_SingleSpeedFields::WaterInletNodeName, *s);
+    }
+  }
+  ///////////////////////////////////////////////////////////////////////////
+
+  ///////////////////////////////////////////////////////////////////////////
+  mo = modelObject.waterOutletNode();
+  if (mo)
+  {
+    s = mo->name();
+    if (s)
+    {
+      idfObject.setString(openstudio::FluidCooler_SingleSpeedFields::WaterOutletNodeName, *s);
+    }
+  }
+  ///////////////////////////////////////////////////////////////////////////
+
+  ///////////////////////////////////////////////////////////////////////////
+  idfObject.setString(openstudio::FluidCooler_SingleSpeedFields::PerformanceInputMethod, modelObject.performanceInputMethod());
+  ///////////////////////////////////////////////////////////////////////////
+
+  ///////////////////////////////////////////////////////////////////////////
+  d = modelObject.designAirFlowRateUfactorTimesAreaValue();
+  if (d)
+  {
+    idfObject.setString(openstudio::FluidCooler_SingleSpeedFields::DesignAirFlowRateUfactorTimesAreaValue, toString(*d));
```

Comment:
@evanweaver You should use setDouble here and in all other places instead of using setString and calling toString(). I don't even know where it is making that call to? Maybe it is converting it to a QString? Either way, not the correct approach.